### PR TITLE
MT3-411 #ready-for-test New repairs page

### PR DESCRIPTION
**Pull request by LBHABEBIC**
New Repairs page in Property Inspection. Can record a repair, take and remove a photo and write a note. Needs to be able to add more than one separate repair.

# What?

# Why?

# Notes

# Next steps

### CODE REVIEW

File: __tests__/features/end-to-end.steps.ts [hunk 1]
```diff
@@ -713,6 +718,23 @@ defineFeature(loadFeature("./end-to-end.feature"), test => {
...
+        })
+        .click();
+      await browser!
+        .waitForEnabledElement({ name: "repair-images" })
+        .sendKeys(processData.property.repair.images[0]);
+      await browser!
+        .waitForEnabledElement({ name: "repair-notes" })
+        .sendKeys(processData.property.repair.notes);
+
+      await browser!.submit();
```
Comment (erbridge) [inline __tests__/features/end-to-end.steps.ts]: Nice! I was expecting to have to explain this to you :).

File: steps/PageTitles.ts [hunk 2]
```diff
@@ -43,6 +42,7 @@ enum PageTitles {
   Pets = "Pets",
   AntisocialBehaviour = "Antisocial behaviour",
   OtherComments = "Other comments",
+  Repairs = "Repairs",
```
Comment (erbridge) [inline steps/PageTitles.ts]: Let's move this back to where it was. The order of this enum helps see the order of the pages.

File: steps/property-inspection/index.ts [hunk 3]
```diff
@@ -31,5 +32,6 @@ export default [
   communalAreas,
   pets,
   antisocialBehaviour,
-  otherComments
+  otherComments,
+  repairs
```
Comment (erbridge) [inline steps/property-inspection/index.ts]: Please insert it in the place where it lives in the journey.

File: steps/property-inspection/repairs.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        value: "Save and continue"
+      }),
+    componentWrappers: [
+      ComponentWrapper.wrapStatic(
+        new StaticComponent({
+          key: "paragraph-1",
+          Component: Paragraph,
+          props: {
+            children: `Residents can contact the Repairs Contact Centre: 020 8356 3691
+            or [Report a repair] online (online only, opens in a new tab).`
```
Comment (erbridge) [inline steps/property-inspection/repairs.tsx]: `[Report a repair]` needs to be a link to the appropriate place.

File: storage/ProcessDatabaseSchema.ts [hunk 5]
```diff
@@ -45,6 +45,11 @@ type ProcessDatabaseSchema = NamedSchema<
           images: string[];
           notes: string;
         };
+        repair: {
+          hasRepair: string;
```
Comment (erbridge) [inline storage/ProcessDatabaseSchema.ts]: Let's name this `needsRepairs` instead.

File: steps/property-inspection/repairs.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      ),
+      ComponentWrapper.wrapDynamic(
+        new DynamicComponent({
+          key: "repair-images",
+          Component: ImageInput,
+          props: {
+            label: "Take photo of repair needed",
+            name: "repair-images",
+            hintText: "" as string | null | undefined,
+            maxCount: 1 as number | null | undefined
```
Comment (erbridge) [inline steps/property-inspection/repairs.tsx]: Let's make this much larger. 10? See if Gill has a reckon on how many, and note the limit like we do for other questions.

**Review by erbridge — COMMENTED** [review]
Let's pluralize the questions and hint text. That way, if we never get to doing multiple repairs, the housing officer can still document what they need to document.

File: steps/property-inspection/repairs.tsx (hunk 6 above)
Comment (LBHABEBIC) [reply steps/property-inspection/repairs.tsx]: Wasn't sure about this as Gill wanted each new picture to be associated with a new note. Damages is max count 5 and garden is 3. happy to do 10 if you think that's a good solution for now.

Comment (erbridge) [reply steps/property-inspection/repairs.tsx]: I'd like to leave this in a state where if we don't get to doing multiple repairs (which is non-trivial), the housing officer can still document what they need to.

File: steps/property-inspection/repairs.tsx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+          props: {
+            label: {
+              value: "Add note about repairs if necessary."
+            } as { id?: string; value: React.ReactNode },
+            name: "repairs-notes"
+          },
+          renderWhen(stepValues: {
+            "has-repairs"?: ComponentValue<ProcessDatabaseSchema, "property">;
+          }): boolean {
+            return stepValues["has-repairs"] === "yes";
```
Comment (erbridge) [inline steps/property-inspection/repairs.tsx]:
```suggestion
            "needs-repairs"?: ComponentValue<ProcessDatabaseSchema, "property">;
          }): boolean {
            return stepValues["needs-repairs"] === "yes";
```

**Review by erbridge — COMMENTED** [review]
Last request: Please can you squash all these commits together for a clean commit history. Grab me or @Gweaton if you need help doing this.

**Pull request merged**